src/picongpu/include/particles/startPosition/RandomImpl.hpp

```diff
@@ -53,8 +53,11 @@ struct RandomImpl
         typedef typename SpeciesType::FrameType FrameType;
 
         mpi::SeedPerRank<simDim> seedPerRank;
-        seed = seedPerRank(GlobalSeed()(), PMacc::traits::GetUniqueTypeId<FrameType, uint32_t>::uid());
-        seed ^= POSITION_SEED;
+        GlobalSeed globalSeed;
+        seed = globalSeed() ^
+               PMacc::traits::GetUniqueTypeId<FrameType, uint32_t>::uid() ^
+               POSITION_SEED;
+        seed = seedPerRank(seed);
 
```

Comment:
pls add the `currentStep` xor (can also be, in principle, now be used at each time step)